asdf_astropy/tags/transform/basic.py

```diff
@@ -0,0 +1,216 @@
+            fixed_nondefaults = {k: f for k, f in model.fixed.items() if f}
+            if fixed_nondefaults:
+                node['fixed'] = fixed_nondefaults
+            bounds_nondefaults = {k: b for k, b in model.bounds.items() if any(b)}
+            if bounds_nondefaults:
+                node['bounds'] = bounds_nondefaults
+
+        return node
+
+    def to_yaml_tree(self, model):
+        raise NotImplementedError("Must be implemented in TransformType subclasses")
+
+    def to_tree(self, model):
+        node = self.to_tree_transform(model)
+        return self._to_tree_base_transform_members(model, node)
+
+    def assert_equal(self, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        assert a.name == b.name
+        # TODO: Assert inverses are the same
+
+
+class IdentityType(TransformType):
+    tags = {"http://asdf-format.org/schemas/transform/identity-2.0.0"}
+    types = {models.Identity}
+
+    def from_yaml_tree(self, node):
+        return models.Identity(node.get('n_dims', 1))
+
+    def to_yaml_tree(self, model):
+        node = {}
+        if model.n_inputs != 1:
+            node['n_dims'] = model.n_inputs
+        return node
+
+    def assert_equal(self, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        TransformType.assert_equal(a, b)
+        assert (isinstance(a, models.Identity) and
+                isinstance(b, models.Identity) and
+                a.n_inputs == b.n_inputs)
+
+
+class ConstantType(TransformType):
+    tags = {"http://asdf-format.org/schemas/transform/constant-2.0.0"}
+    supported_versions = ['1.0.0', '1.1.0', '1.2.0', '1.3.0', '1.4.0', '2.0.0']
+    types = {models.Const1D, models.Const2D}
+
+    def from_yaml_tree(self, node):
+        if self.version < AsdfVersion('1.4.0'):
+            # The 'dimensions' property was added in 1.4.0,
+            # previously all values were 1D.
+            return models.Const1D(node['value'])
+        elif node['dimensions'] == 1:
+            return models.Const1D(node['value'])
+        elif node['dimensions'] == 2:
+            return models.Const2D(node['value'])
+
+    def to_yaml_tree(self, model):
+        if self.version < AsdfVersion('1.4.0'):
+            if not isinstance(model, models.Const1D):
+                raise ValueError(
+                    f'constant-{self.version} does not support models with > 1 dimension')
+            return {
+                'value': _parameter_to_value(model.amplitude)
+            }
+        else:
+            if isinstance(model, models.Const1D):
+                dimension = 1
+            elif isinstance(model, models.Const2D):
+                dimension = 2
+            return {
+                'value': _parameter_to_value(model.amplitude),
+                'dimensions': dimension
+            }
+
+
+class UnitsMappingType(asdf.AsdfConverter):
+    tags = {"http://asdf-format.org/schemas/transform/units_mapping-2.0.0"}
```

Comment:
This transform schema was deemed to be too astropy-specific for the general use transforms, so it's actually still living in astropy with URI `http://astropy.org/schemas/astropy/transform/units_mapping-1.0.0`.

Reply:
Can a third party implementation of the standard implement only part of the schemas? If so, perhaps we should keep all schemas in one place and let people decide what they want to implement. I think it's something we should discuss further.

Reply:
That sounds like a good idea to me, there are so many transforms that the prospect of implementing them all is intimidating.

I wonder if we should separate them into namespaces to guide people on implementing a minimal set?  `transform/core` and others.

Reply:
This worries me as well. I don't think people should be forced to implement them all (e.g., all the projection transform)